components/eam/src/chemistry/mozart/mo_chm_diags.F90

```diff
@@ -752,6 +756,17 @@ subroutine chm_diags( lchnk, ncol, vmr, mmr, rxt_rates, invariants, depvel, depf
        call outfld( 'Mass_soa', mass_soa(:ncol,:,:),ncol,lchnk)
     end if
     endif
+
+#if defined(CLDERA_PROFILING)
+    call cldera_set_field_part_data("Mass_so4" ,lchnk-begchunk+1,mass_so4(:ncol,:,:))
```

Comment:
Is there a problem defining this now that it is broken into tagged values?

Reply:
This one is trickier because I think it uses the same data allocation. Therefore, Mass_so4 is probably just the first tagged value which is misleading. I'll move this to an else like it's done here:
https://github.com/sandialabs/CLDERA-E3SM/blob/f58d8e7b866513d64d4c8c9ebb381ef2bffbd4f1/components/eam/src/chemistry/mozart/mo_chm_diags.F90#L747-L748

Reply:
I think that is a good approach. It is better for backwards compatibility with the non-tagging code as well.